### PR TITLE
[WIP] Introduce dynamic index patterns

### PR DESCRIPTION
**Pull request by ruflin**
With the introduction of `migration.enabled` config option the index pattern must generated on the fly to skip migration alias if needed.

Currently this PR generates the index pattern as a file in the Kibana directory and then it reads in the data as before. This so far seems to work but as not ideal. The fieldsYML data should be loaded directly from the go fields and the file should never be generated on disk.

TODO

* Stop generation of index pattern on make update
* Make sure packaging works as expected withut index pattern
* Remove directory
* Introduce command to export index pattern -> needed?

### CODE REVIEW

File: libbeat/dashboards/dashboards.go [hunk 1]
```diff
@@ -59,6 +62,29 @@ func ImportDashboards(
 
 	return setupAndImportDashboardsViaKibana(ctx, hostname, kibanaConfig, &dashConfig, msgOutputter)
 
+	// Generate index pattern
+	version, _ := common.NewVersion("7.0.0") // TODO: dynamic version
+	//beatVersion, _ := common.NewVersion("7.0.0") // TODO: dynamic version
+	//indexP, _ := esConfig.String("index", 0)
```
Comment (ruflin) [inline libbeat/dashboards/dashboards.go]: @urso I think this also ties into your current work around indexing changes. Where should the config which index pattern should be used go for Kibana setup part?

File: CHANGELOG-developer.next.asciidoc [hunk 2]
```diff
@@ -20,6 +20,8 @@ The list below covers the major changes between 7.0.0-alpha2 and master only.
 
 ==== Breaking changes
 
+- Remove support for loading dashboards into Elastic Stack version 5.x. {pull}10451[10451]
```
Comment (ruflin) [inline CHANGELOG-developer.next.asciidoc]: Taken over for 10451 as it was missing there.

File: libbeat/dashboards/importer.go [hunk 3]
```diff
@@ -312,7 +304,7 @@ func (imp Importer) ImportKibanaDir(dir string) error {
 
 	check := []string{}
 	if !imp.cfg.OnlyDashboards {
-		check = append(check, "index-pattern")
+		imp.loader.ImportIndex()
```
Comment (ruflin) [inline libbeat/dashboards/importer.go]: This breaks current behaviour in that it makes it impossible to load an index pattern from a file and requires the one from Beats.

**Comment by simitt** [comment]
@ruflin APM does not support loading the Kibana index pattern via the Server, but it is loaded directly via Kibana. Since the whole ES template and mapping logic is in the APM Server we still rely on the index pattern being created with `make update` so we can sync it and ensure we bundle the latest index pattern with Kibana. 
For this requirements, please ensure to still provide an option to generate and export the Kibana index pattern. In case you are aiming for sending the index pattern to ES during ingestion time (which I understand the _dynamic_ part implies) please also provide an option to disable this.

**Comment by ruflin** [comment]
Closing this in favor of https://github.com/elastic/beats/pull/10478 as I had to take a bit a different approach because the generation has to stay around but can be an exporter and also the old loading mechanism for zip file has to stay around for now.

**Pull request closed without merging**